shell/platform/embedder/embedder.h

```diff
@@ -2582,9 +2582,23 @@ FlutterEngineResult FlutterEngineRunInitialized(
 //------------------------------------------------------------------------------
 /// @brief      Adds a view.
 ///
-///             This is an asynchronous operation. The view should not be used
-///             until the |add_view_callback| is invoked with an `added` of
-///             `true`.
+///             This operation notifies the engine to allocate resources and
+///             inform Dart about the view, and on success, schedules a new
+///             frame. Finally, it invokes |info.add_view_callback| with whether
+///             the operation is successful.
+///
+///             This operation is asynchronous; avoid using the view until
+///             |info.add_view_callback| returns true. The embedder should
+///             prepare resources in advance but be ready to clean up on
+///             failure.
+///
+///             The callback is called on a different thread managed by the
+///             engine. The embedder must re-thread if needed.
+///
+///             Do not use for implicit views, which are added internally during
+///             initialization. Adding the implicit view ID or an existing view
```

Comment:
Attempting to add the implicit view ID will return a `kInvalidArguments` immediately. The callback will not be invoked.